### DISCUSSION TITLE
Multi-action Query

### DISCUSSION THREAD

**Discussion by BigBadAlien** [klis87/redux-saga-requests#442] [General]
Any reason to not support it?

```
<Query type={[ACTION1, ACTION2, ...]}
```

Or otherwise, how to deal with the nesting of Mutation and Query?

**Comment by klis87** [comment]
Nesting is solved by hooks, like `useQuery` and `useMutation`. Actually, probably all render prop components will be removed in the next version, as mentioned in https://github.com/klis87/redux-requests/issues/417 . Why? Because those are very easy to create on the user land, as well as HOCs, and maintaining them is a cost, as well as they add to the bundle size, which is bad for marketing. 

Anyway, please pay attention that current `Query` is much less powerful than `useQuery`, which has things like auto fetching and so on. On the other hand, you can say that `Query` is quite opinionated, it has many props which could be against someone's taste, in the contrast to `useQuery`. Then, if one needs a render prop component, one can just create one's own based on `useQuery`, however see fit.

Regarding supporting array of actions, I thought about it, but it is not as easy, as we have many other properties besides `type`, I might add something like `useQueries`, but I would need to think how it would look like and whether it is worth it.

**Reply by BigBadAlien** [reply]
`useQueries` is not really interesting because the nesting problem is not typical for hooks. Ok, got it, makes sense to leave the Query and Mutation just as an example and remove it from the main codebase. Thank you very much
